pkg/metrics/metrics.go

```diff
@@ -250,3 +255,38 @@ func updateServiceMonitors(cfg *rest.Config, namespace string) error {
 	}
 	return nil
 }
+
+// removeInvalidResources deletes any stale resources that could be left behind during the
+// upgrade process due to the renaming of resources between current and previous operator version
+func removeInvalidResources(ctx context.Context, cfg *rest.Config, namespace string) error {
+	invalidResourceName := "windows-machine-config-operator-metrics"
+
+	oclient, err := k8sclient.NewForConfig(cfg)
```

Comment:
These clients seem to be getting created in multiple places. Maybe this calls for a configuration metrics object?

Reply:
created a struct around them